src/metabase/sync/interface.clj

```diff
@@ -22,21 +22,22 @@
 
 (def TableMetadataField
   "Schema for a given Field as provided in `describe-table`."
-  {:name                               su/NonBlankString
-   :database-type                      (s/maybe su/NonBlankString) ; blank if the Field is all NULL & untyped, i.e. in Mongo
-   :base-type                          su/FieldType
-   :database-position                  su/IntGreaterThanOrEqualToZero
-   (s/optional-key :semantic-type)     (s/maybe su/FieldSemanticOrRelationType)
-   (s/optional-key :effective-type)    (s/maybe su/FieldType)
-   (s/optional-key :coercion-strategy) (s/maybe su/CoercionStrategy)
-   (s/optional-key :field-comment)     (s/maybe su/NonBlankString)
-   (s/optional-key :pk?)               s/Bool
-   (s/optional-key :nested-fields)     #{(s/recursive #'TableMetadataField)}
-   (s/optional-key :nfc-path)          [s/Any]
-   (s/optional-key :custom)            {s/Any s/Any}
-   (s/optional-key :database-required) s/Bool
+  {:name                                       su/NonBlankString
+   :database-type                              (s/maybe su/NonBlankString) ; blank if the Field is all NULL & untyped, i.e. in Mongo
+   :base-type                                  su/FieldType
+   :database-position                          su/IntGreaterThanOrEqualToZero
+   (s/optional-key :semantic-type)             (s/maybe su/FieldSemanticOrRelationType)
+   (s/optional-key :effective-type)            (s/maybe su/FieldType)
+   (s/optional-key :coercion-strategy)         (s/maybe su/CoercionStrategy)
+   (s/optional-key :field-comment)             (s/maybe su/NonBlankString)
+   (s/optional-key :pk?)                       s/Bool
+   (s/optional-key :nested-fields)             #{(s/recursive #'TableMetadataField)}
+   (s/optional-key :nfc-path)                  [s/Any]
+   (s/optional-key :custom)                    {s/Any s/Any}
+   (s/optional-key :database-auto-incremented) s/Bool
```

Comment:
Remind me why this is optional? It's a non-null field, like base-type and database-position. And `database_required` too?

Reply:
I think they're optional because we the return of  [fallback-fields-metadata-from-select-query](https://github.com/metabase/metabase/blob/24500d8f8733b8bc37fe0eeca1d161023243ace8/src/metabase/driver/sql_jdbc/sync/describe_table.clj#L72) match the schema too.

I'm not really sure tho, just following the convention here.

Reply:
anw, updated to make this key required: 73e6e96158